src/index.ts

```diff
@@ -1,43 +1,83 @@
- *  the min and max value of the parameter (search space) and the step size to approximate the jacobian matrix respectively. Those options could be a number,
- *  array of numbers, callback, or array of callbacks. Each kind of shape has default parameters so it could be undefined.
- * @param {object} [options.optimization.parameters.x] - options for x parameter.
- * @param {number|callback|array<number|callback>} [options.optimization.parameters.x.init] - definition of the starting point of the parameter (the guess),
- *  if it is a callback the method pass the peak as the unique input, if it is an array the first element define the guess of the first peak and so on.
- * @param {number|callback|array<number|callback>} [options.optimization.parameters.x.min] - definition of the lower limit of the parameter,
- *  if it is a callback the method pass the peak as the unique input, if it is an array the first element define the min of the first peak and so on.
- * @param {number|callback|array<number|callback>} [options.optimization.parameters.x.max] - definition of the upper limit of the parameter,
- *  if it is a callback the method pass the peak as the unique input, if it is an array the first element define the max of the first peak and so on.
- * @param {number|callback|array<number|callback>} [options.optimization.parameters.x.gradientDifference] - definition of  the step size to approximate the jacobian matrix of the parameter,
- *  if it is a callback the method pass the peak as the unique input, if it is an array the first element define the gradientDifference of the first peak and so on.
- * @param {object} [options.optimization.options = {}] - options for the specific kind of algorithm.
- * @param {number} [options.optimization.options.timeout] - maximum time running before break in seconds.
- * @param {number} [options.optimization.options.damping=1.5]
- * @param {number} [options.optimization.options.maxIterations=100]
- * @param {number} [options.optimization.options.errorTolerance=1e-8]
- * @returns {object} - A object with fitting error and the list of optimized parameters { parameters: [ {x, y, width} ], error } if the kind of shape is pseudoVoigt mu parameter is optimized.
+ *
+ * @param data - An object containing the x and y data to be fitted.
+ * @param peakList - A list of initial parameters to be optimized. e.g. coming from a peak picking [{x, y, width}].
+ * @param options - Options.
+ * @returns - An object with fitting error and the list of optimized parameters { parameters: [ {x, y, width} ], error } if the kind of shape is pseudoVoigt mu parameter is optimized.
  */
-
 export function optimize(
   data: DataXY<DoubleArray>,
   peakList: Peak1D[],
-  options: OptimizeOptions,
+  options: {
+    /**
+     * kind of shape used for fitting
+     **/
+    shape?:
+      | Shape1D
+      | {
```

Comment:
I think we should only allow a Shape1D

Reply:
In the test file though we have the following though:

optimize(modifiedData, peakList, {
      shape: { kind: 'lorentzian' }
});